core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -247,10 +248,19 @@ class UnifiedLog(@volatile var logStartOffset: Long,
                  @volatile private var _topicId: Option[Uuid],
                  val keepPartitionMetadataFile: Boolean,
                  val remoteStorageSystemEnable: Boolean = false,
-                 remoteLogManager: Option[RemoteLogManager] = None) extends Logging with KafkaMetricsGroup {
+                 remoteLogManager: Option[RemoteLogManager] = None) extends Logging {
 
   import kafka.log.UnifiedLog._
 
+  private val metricsGroup = new KafkaMetricsGroup(this.getClass) {
+    // For compatibility, metrics are defined to be under `Log` class
+    override def metricName(name: String, tags: util.Map[String, String]): MetricName = {
+      val pkg = getClass.getPackage
+      val pkgStr = if (pkg == null) "" else pkg.getName
```

Comment:
This seems unnecessary, we know the package is not null.

Reply:
Sure, removed.